api.py

```diff
@@ -0,0 +1,116 @@
+import os
+import threading
+import enum
+from time import sleep
+from subprocess import Popen, DEVNULL
+
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+
+from pytube import YouTube
+from pytube.exceptions import AgeRestrictedError, VideoUnavailable, RegexMatchError
+
+
+class State(enum.Enum):
+    INTERLUDE = "interlude"
+    PLAYING = "playing"
+
+app = FastAPI()
+interlude_command = ['ffmpeg', '-re', '-stream_loop', '-1', '-i', "./interlude.mp4", '-vf', 
+                     'scale=426:240', '-c:v', 'libx264', 
+                     '-preset', 'veryfast', '-tune', 'zerolatency', 
+                     '-c:a', 'aac', '-ar', '44100', '-f', 'flv', 
+                     'rtmp://localhost:1935/live/mystream']
+process_dict = {}
+process_dict[State.INTERLUDE] = Popen(interlude_command, stdout=DEVNULL, stdin=DEVNULL, stderr=DEVNULL)
+current_video_data = {}
+
+# Ensure video folder exists
+if not os.path.exists("./videos"):
+   os.makedirs("./videos")
+
+# Enable CORS
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+def download_and_play(url:str):
+    # Download video
+    video = YouTube(url)
+    video = video.streams.get_by_resolution("360p")
+    videoToPlay = video.default_filename
+    # Download video is not already downloaded
+    if (videoToPlay not in os.listdir("./videos")):
+        video.download("./videos/")
+        # Wait until video is downloaded
+        while videoToPlay not in os.listdir("./videos"):
+            sleep(0.5)
+    # Stop interlude
+    interlude_process = process_dict.pop(State.INTERLUDE)
+    interlude_process.terminate()
+    interlude_process.wait()
+    # Start streaming video
+    process_dict[State.PLAYING] = Popen(['ffmpeg', '-re', '-i', f"./videos/{videoToPlay}", 
```

Comment:
make this its own function like

```py
def create_ffmpeg_stream(video_path: str, video_type: State):
    command = ['ffmpeg', '-re', '-stream_loop', '-1', '-i', video_path, '-vf', 
                     'scale=426:240', '-c:v', 'libx264', 
                     '-preset', 'veryfast', '-tune', 'zerolatency', 
                     '-c:a', 'aac', '-ar', '44100', '-f', 'flv', 
                     'rtmp://localhost:1935/live/mystream']
    process_dict[video_type] = Popen(
        command,
        stdout=subprocess.DEVNULL,
        stdin=subprocess.DEVNULL,
        stderr=subprocess.DEVNULL,
    )
```

then, starting the various processes becomes trivial like
```py
create_ffmpeg_stream('./path/to/interlude', State.INTERLUDE)

# then somewhere else
create_ffmpeg_stream('./path/to/downloaded/file', State.PLAYING)
```